problemtools/verifyproblem.py

```diff
@@ -1221,35 +1232,60 @@ def __init__(self, problem):
     def __str__(self):
         return 'submissions'
 
-    def check_submission(self, sub, args, expected_verdict, timelim_low, timelim_high):
+    def check_submission(self, sub, args, expected_verdict, timelim, timelim_low, timelim_high):
         desc = '%s submission %s' % (expected_verdict, sub)
-        (result1, result2) = self._problem.testdata.run_submission(sub, args, timelim_low, timelim_high)
-
-        if result1.verdict != result2.verdict:
-            r1, r2 = result1.verdict, result2.verdict
-            self.warning('%s sensitive to time limit: limit of %s secs -> %s, limit of %s secs -> %s' % (desc, timelim_low, r1, timelim_high, r2))
-
-        if result1.verdict == expected_verdict:
+        partial = False
+        if expected_verdict == 'PAC':
+            # For partially accepted solutions, use the low timelim instead of the real one,
+            # to make sure we have margin in both directions.
+            expected_verdict = 'AC'
+            partial = True
+            timelim = timelim_low
```

Comment:
As far as I understand, the purpose of this is to give a warning if, when running the submission with half the time limit (where "half"=whatever the 1/safety margin is, 0.5 by default), it gives a different result than when running with twice the time limit (where "twice"=...)?  Why is this necessary, and what makes PAC different from other verdicts that it is warranted for PAC but not for other verdicts?

Reply:
The idea is to make sure that if you include a solution in partially_accepted/, making it slightly slower or slightly faster should not affect its score.

For e.g. a TLE solution, we check that making it slightly faster doesn't make it get accepted, but checking that making it slower doesn't affect its score is pointless since it scores 0.

Arguably you'd want to be able to say "I wrote a slow Python solution, please ensure its score doesn't change when making it 1.5 times slower or 20 times faster", but we're going with only time_safety_margin for now in the hope that that's good enough in practice.

Reply:
To clarify (not sure if it's needed). We believe that for completeness you would actually need to specify an upper as well as a lower safety margin. It's not (at all) clear to me that these must always be the same. OTOH, we suspect that in practice having the lower be the same (or the inverse, depending on how you look at it) as the upper (i.e. the only one we have) is good enough.

Also, if it turns out that you *do* need to specify a lower margin once in a while, then what we are using now is certainly a very reasonable default, so it will be safe to introduce it later.